plutus-tx-plugin/src/PlutusTx/Compiler/Builtins.hs

```diff
@@ -244,7 +245,7 @@ defineBuiltinTerm ann name term = do
     ghcId <- GHC.tyThingId <$> getThing name
     var <- compileVarFresh ann ghcId
     -- See Note [Builtin terms and values]
-    let strictness = if PIR.isPure (const PIR.NonStrict) term then PIR.Strict else PIR.NonStrict
+    let strictness = if PIR.isPure PLC.currentVerDefaultFun (const PIR.NonStrict) term then PIR.Strict else PIR.NonStrict
```

Comment:
suspect. This should be configurable and be in the compilation context. Or in the types.